### PR TITLE
test: fix privilege checks and make persistPath testable 

### PR DESCRIPTION
**Pull request by yuvraj-kolkar17**
**What type of PR is this?**
This PR fixes test failures related to privilege requirements and makes the test suite more flexible. It also adds additional tests.

/kind bug
/kind enhancement

**What this PR does / why we need it**:

1. **Privilege-related test failures**: `TestHandleKmeshManage` required root privileges for namespace operations but failed without proper handling, blocking non-privileged test execution
2. **Path immutability**: `persistPath` was defined as a constant, preventing tests from using temporary directories and causing permission errors
3. **Insufficient test coverage**: Missing test cases for edge cases and error scenarios

# Changes
- **pkg/utils/hash_name.go**: Changed `persistPath` from `const` to `var` to allow testing with temporary directories
- **pkg/utils/hash_name_test.go**: Added test setup helpers (`setupTestPersistPath`, `restorePersistPath`) for proper test isolation
- **pkg/utils/enroll_test.go**: Added privilege check to `TestHandleKmeshManage` to skip when not running as root

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/utils/hash_name.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-const (
+var (
```
Comment (LiZhenCheng9527) [inline pkg/utils/hash_name.go:29]: Revert!
